WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/details/ThreatDetailsViewModel.kt

```diff
@@ -1,44 +1,147 @@
     private var isStarted = false
     private var threatId: Long = 0
 
     private val _uiState: MutableLiveData<UiState> = MutableLiveData()
     val uiState: LiveData<UiState> = _uiState
 
+    private val _snackbarEvents = MediatorLiveData<Event<SnackbarMessageHolder>>()
+    val snackbarEvents: LiveData<Event<SnackbarMessageHolder>> = _snackbarEvents
+
+    private val _navigationEvents = MediatorLiveData<Event<ThreatDetailsNavigationEvents>>()
+    val navigationEvents: LiveData<Event<ThreatDetailsNavigationEvents>> = _navigationEvents
+
     fun start(threatId: Long) {
         if (isStarted) {
             return
         }
         isStarted = true
         this.threatId = threatId
+        site = requireNotNull(selectedSiteRepository.getSelectedSite())
         getData()
     }
 
     private fun getData() {
         viewModelScope.launch {
             val model = getThreatModelUseCase.get(threatId)
-            model?.let { updateUiState(it) }
+            model?.let { updateUiState(buildContentUiState(it)) }
+        }
+    }
+
+    private fun ignoreThreat() {
+        viewModelScope.launch {
+            disableThreatActionButtons(true)
+            when (ignoreThreatUseCase.ignoreThreat(site.siteId, threatId)) {
+                is IgnoreThreatState.Success -> {
+                    updateSnackbarMessageEvent(UiStringRes(R.string.threat_ignore_success_message))
+
+                    withContext(bgDispatcher) { delay(DELAY_MILLIS) }
+                    updateNavigationEvent(ShowUpdatedScanState)
+                }
+
+                is IgnoreThreatState.Failure.NetworkUnavailable -> {
+                    disableThreatActionButtons(false)
+                    updateSnackbarMessageEvent(UiStringRes(R.string.error_generic_network))
+                }
+
+                is IgnoreThreatState.Failure.RemoteRequestFailure -> {
+                    disableThreatActionButtons(false)
+                    updateSnackbarMessageEvent(UiStringRes(R.string.threat_ignore_error_message))
+                }
+            }
+        }
+    }
+
+    private fun onFixThreatButtonClicked() { // TODO ashiagr to be implemented
+    }
+
+    private fun onIgnoreThreatButtonClicked() {
+        updateNavigationEvent(
+            OpenThreatActionDialog(
+                title = UiStringRes(R.string.threat_ignore),
+                message = UiStringText(
+                    htmlMessageUtils
+                        .getHtmlMessageFromStringFormatResId(
+                            R.string.threat_ignore_warning,
+                            "<b>${site.name ?: resourceProvider.getString(R.string.scan_this_site)}</b>"
+                        )
+                ),
+                okButtonAction = this@ThreatDetailsViewModel::ignoreThreat
+            )
+        )
+    }
+
+    private fun onGetFreeEstimateButtonClicked() { // TODO ashiagr to be implemented
+    }
+
+    private fun disableThreatActionButtons(disable: Boolean) {
```

Comment:
💡 Idea: We might want to consider renaming this to "updateThreatActionButtons" or similar. It feels a bit weird that the method is called "disable" but it actually enables the items when you pass "false". Wdyt?

Reply:
Sounds good, updated in 8429c5a